citesphere/src/main/java/edu/asu/diging/citesphere/web/user/SyncItemController.java

```diff
@@ -1,26 +1,77 @@
 package edu.asu.diging.citesphere.web.user;
 
+import java.util.List;
+import java.util.Optional;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.core.Authentication;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 
 import edu.asu.diging.citesphere.core.exceptions.CannotFindCitationException;
 import edu.asu.diging.citesphere.core.exceptions.GroupDoesNotExistException;
 import edu.asu.diging.citesphere.core.model.IUser;
+import edu.asu.diging.citesphere.core.model.bib.ICitationGroup;
+import edu.asu.diging.citesphere.core.model.bib.impl.CitationGroup;
+import edu.asu.diging.citesphere.core.repository.bib.CitationGroupRepository;
 import edu.asu.diging.citesphere.core.service.ICitationManager;
+import edu.asu.diging.citesphere.core.zotero.IZoteroConnector;
 
 @Controller
 public class SyncItemController {
     
     @Autowired
     private ICitationManager citationManager;
+    
+    @Autowired
+    private IZoteroConnector zoteroConnector;
+    
+    @Autowired
+    private CitationGroupRepository groupRepository;
 
     @RequestMapping(value="/auth/group/{zoteroGroupId}/items/{itemId}/sync")
     public String getItem(Authentication authentication, Model model, @PathVariable("zoteroGroupId") String zoteroGroupId, @PathVariable("itemId") String itemId) throws GroupDoesNotExistException, CannotFindCitationException {
         citationManager.updateCitationFromZotero((IUser)authentication.getPrincipal(), zoteroGroupId, itemId);
         return "redirect:/auth/group/{zoteroGroupId}/items/{itemId}";
     }
+    
+    @RequestMapping(value = {"/auth/group/{zoteroGroupId}/items/sync", "/auth/group/{zoteroGroupId}/collection/{collectionId}/items/sync"})
+    public String getCitationsList(Authentication authentication, Model model,
```

Comment:
needs to go in its own controller

Reply:
Should I create a new controller 'SyncGroupItemsController' or move it in GroupItemsController?

Reply:
new controller